crates/block-producer/src/server.rs

```diff
@@ -261,11 +290,34 @@ impl BlockProducerRpcServer {
         let inputs = self.store.get_tx_inputs(&tx).await.map_err(VerifyTxError::from)?;
 
         // SAFETY: we assume that the rpc component has verified the transaction proof already.
+        let tx_id = tx.id();
+        let tx_nullifiers: Vec<Nullifier> = tx.nullifiers().collect();
+
         let tx = AuthenticatedTransaction::new(tx, inputs)?;
 
-        self.mempool.lock().await.lock().await.add_transaction(tx).map(|block_height| {
-            SubmitProvenTransactionResponse { block_height: block_height.as_u32() }
-        })
+        // Launch a task for updating the mempool, and send the update to the network transaction
+        // builder
+        let (submit_tx_response, _) = tokio::join!(
+            async {
+                self.mempool.lock().await.lock().await.add_transaction(tx).map(|block_height| {
+                    SubmitProvenTransactionResponse { block_height: block_height.as_u32() }
+                })
+            },
+            async {
+                let mut ntx_client = self.ntx_client.clone();
+                if let Err(err) =
+                    ntx_client.update_network_notes(tx_id, tx_nullifiers.into_iter()).await
+                {
+                    warn!(
+                        target: COMPONENT,
+                        message = %err,
+                        "error submitting network notes updates to ntx builder"
+                    );
+                }
+            },
+        );
```

Comment:
This tries to update the mempool and send the transaction to the NTB at the same time - right? If so, I wonder if we should actually do this sequentially. That is, add transaction to the mempool and if it was added without errors, only then send it to NTB.

Reply:
I changed this to work sequentially on the execution PR, but forgot to change it back here. Will do so now.